### PR TITLE
Force runtime initialization of BlockSource$ThreadedCachingBlockSource

### LINKED ISSUE #27101: Native + Oracle JDBC + Quarkus NativeImageBuildStep fails
### Describe the bug

Quarkus Upstream
Extension: `quarkus-jdbc-oracle / quarkus-reactive-oracle-client`

Error:

```
Error: Classes that should be initialized at run time got initialized during image building:
 oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource 
```

Stacktrace:

```

[1/7] Initializing...                                                                                    (9.3s @ 0.39GB)
 Version info: 'GraalVM 22.1.0.0-Final Java 11 Mandrel Distribution'
 C compiler: gcc (redhat, x86_64, 8.5.0)
 Garbage collector: Serial GC
 6 user-provided feature(s)
  - io.quarkus.hibernate.validator.runtime.DisableLoggingFeature
  - io.quarkus.jdbc.postgresql.runtime.graal.SQLXMLFeature
  - io.quarkus.runner.Feature
  - io.quarkus.runtime.graal.DisableLoggingFeature
  - io.quarkus.runtime.graal.ResourcesFeature
  - oracle.nativeimage.NativeImageFeature
The bundle named: com/sun/rowset/RowSetResourceBundle, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
[2/7] Performing analysis...  [***********]                                                             (69.9s @ 2.25GB)
  19,926 (92.22%) of 21,608 classes reachable
  37,342 (60.75%) of 61,470 fields reachable
 115,287 (61.57%) of 187,252 methods reachable
     529 classes, 1,059 fields, and 3,470 methods registered for reflection
      76 classes,   113 fields, and    55 methods registered for JNI access
[3/7] Building universe...                                                                               (5.8s @ 2.38GB)
[4/7] Parsing methods...      [*****]                                                                   (21.3s @ 1.96GB)
[5/7] Inlining methods...     [*****]                                                                   (21.0s @ 2.77GB)
[6/7] Compiling methods...    [*********]                                                               (80.6s @ 3.02GB)
To see how the classes got initialized, use --trace-class-initialization=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource,oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser

[7/7] Creating image...                                                                                  (0.0s @ 2.65GB)
Error: Classes that should be initialized at run time got initialized during image building:
 oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'BlockSource$ThreadedCachingBlockSource.class'). To see why oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource got initialized use --trace-class-initialization=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource
oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'BlockSource$ThreadedCachingBlockSource$BlockReleaser.class'). To see why oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser got initialized use --trace-class-initialization=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser

com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'BlockSource$ThreadedCachingBlockSource.class'). To see why oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource got initialized use --trace-class-initialization=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource
oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'BlockSource$ThreadedCachingBlockSource$BlockReleaser.class'). To see why oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser got initialized use --trace-class-initialization=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser

        at com.oracle.svm.core.util.UserError.abort(UserError.java:72)
        at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.checkDelayedInitialization(ConfigurableClassInitialization.java:560)
        at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.afterImageWrite(ClassInitializationFeature.java:306)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$doRun$8(NativeImageGenerator.java:701)
        at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:74)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:701)
        at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:515)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:615)
------------------------------------------------------------------------------------------------------------------------
                       39.2s (17.3% of total time) in 164 GCs | Peak RSS: 4.90GB | CPU load: 6.48
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /project/vertx-sql-1.0.0-SNAPSHOT-runner (executable)
 /project/vertx-sql-1.0.0-SNAPSHOT-runner.build_artifacts.txt
========================================================================================================================
Failed generating 'vertx-sql-1.0.0-SNAPSHOT-runner' after 3m 45s.
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```
Workaround: 
You could add the following native arguments to your application properties:
```
quarkus.native.additional-build-args=--initialize-at-run-time=oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource
```

### Expected behavior

Build succeed without any extra arguments

### Actual behavior

BUILD FAILURE

### How to Reproduce?

Reproducer: 

```
git clone https://github.com/quarkus-qe/quarkus-test-suite.git
```

```
mvn clean verify -Dnative -Dall-modules -pl sql-db/vertx-sql -Dit.test=OracleHandlerIT
```

### Output of `uname -a` or `ver`

Linux localhost.localdomain 5.18.13-100.fc35.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jul 22 14:20:24 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 11.0.14 2022-01-18 OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9) OpenJDK 64-Bit Server VM Temurin-11.0.14+9 (build 11.0.14+9, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #27101

### CODE REVIEW

File: extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java [hunk 1]
```diff
@@ -26,4 +28,10 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, "com.sun.jndi.ldap.LdapCtxFactory"));
     }
 
+    @BuildStep
+    void runtimeInitialization(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClass) {
+        runtimeInitializedClass
+                .produce(new RuntimeInitializedClassBuildItem("oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource"));
```
Comment (pjgg) [inline extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java:34]: Maybe I made a mistake in my original issue description but the class that needs to be registered by reflection is: `oracle.jdbc.driver.BlockSource$ThreadedCachingBlockSource$BlockReleaser`

Comment (geoand) [reply extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java:34]: Pushed an update. Can you test it please?

Comment (pjgg) [reply extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java:34]: These changes are working!

**Review by pjgg — APPROVED** [review]
LGTM

File: extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java (hunk 1 above)
Comment (geoand) [reply extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleNativeImage.java:34]: Thanks for checking!

**Review by gsmet — APPROVED** [review]

**Pull request merged**